### PR TITLE
.github: workflows: Add C++ linter

### PR DESCRIPTION
**Pull request by melvo**
PR check list:
- [x] Document your code
- [x] Add `\since QXmpp 1.X`, `QXMPP_EXPORT`
- [x] Fix doxygen warnings (see log when building with `-DBUILD_DOCUMENTATION=ON`)
- [x] Update `doc/doap.xml`
- [x] Add unit tests
- [x] Format the code: Run `clang-format -i src/<edited-file(s)> tests/<edited-file(s)>`

### CODE REVIEW

**Comment by melvo** [comment]
The [action](https://github.com/cpp-linter/cpp-linter-action) does not support `clang-format-15` (only `clang-format-15`) but the [linting tool](https://github.com/cpp-linter/clang-tools) does.

**Comment by lnjX** [comment]
Currently doesn't seem to work.
```
Performing checkup on src/client/QXmppCarbonManagerV2.cpp
  INFO:CPP Linter:Running "clang-tidy-14 --export-fixes=clang_tidy_output.yml -checks=boost-*,bugprone-*,performance-*,readability-*,portability-*,modernize-*,clang-analyzer-*,cppcoreguidelines-* --extra-arg= src/client/QXmppCarbonManagerV2.cpp"
  DEBUG:CPP Linter:Output from clang-tidy:
  
  INFO:CPP Linter:Running "clang-format-14 -style=file --output-replacements-xml src/client/QXmppCarbonManagerV2.cpp"
  DEBUG:CPP Linter:clang-format-14 raised the following error(s):
  /home/runner/work/qxmpp/qxmpp/.clang-format:27:1: error: unknown key 'InsertBraces'
  InsertBraces: true
  ^~~~~~~~~~~~
  Error reading /home/runner/work/qxmpp/qxmpp/.clang-format: Invalid argument
  ```

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -55,3 +55,19 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cpp-linter/cpp-linter-action@v2
+        id: linter
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          style: file
+          version: 14
```
Comment (melvo) [inline .github/workflows/tests.yml:68]:
```suggestion
          version: 15
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -55,3 +55,19 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cpp-linter/cpp-linter-action@v2
+        id: linter
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          style: file
+          version: 15
```
Comment (melvo) [inline .github/workflows/tests.yml:68]:
```suggestion
          version: 14
```

**Comment by melvo** [comment]
`clang-format-15` is supported now. Result: Only [`src/client/QXmppFileSharingManager.cpp`](https://github.com/qxmpp-project/qxmpp/pull/491/files#diff-e8fbcf424898cae6fc11412c52bbd57a6bd05561b745e4e07134155327fd402d) seems to need adjustments or maybe `.clang-format`.

**Comment by melvo** [comment]
@lnjX This can be reviewed now and should not be a draft anymore.

**Review by lnjX — APPROVED** [review]

**Pull request merged**